lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2722,6 +2730,31 @@ class ConvertAtenIntTensorOp : public OpConversionPattern<AtenIntTensorOp> {
 };
 } // namespace
 
+
+namespace {
+class ConvertAtenFill_ScalarOp : public OpConversionPattern<AtenFill_ScalarOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenFill_ScalarOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op->getLoc();
+    Value self = adaptor.self();
+    Value initVal = adaptor.value();
```

Comment:
Need `convertScalarToDtype` first to make the type match. See fillOp verification code here https://github.com/llvm/llvm-project/blob/cf36ab1d6c39e80a70b5a1dc80120cccccb5301c/mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp#L482.

It would be good to add a e2e test case for cases where the scalar has different types as the tensor.

Reply:
I see, I just missed that. I have added it with a new test case. Thanks.